### DISCUSSION TITLE
Remove Extreme Speed Preset (LCM) - Yes or No?

### DISCUSSION THREAD

**Discussion by mashb1t** [lllyasviel/Fooocus#3094] [Polls]
Pros:
* prevent users from accidentally generating images in lower quality as LCM is quite outdated
* reduce legacy code and promote more recent and higher quality methods like Hyper-SD

Cons:
* users can't regenerate images from image log anymore, which have been generated with Extreme Speed

What do you think? Happy to hear your opinion.

**Comment by mashb1t** [comment]
Alternatives:
* replace performance option group with select element (dropdown)
* use different performance with same label "Extreme Speed" (might lead to confusion, see [#3090](https://github.com/lllyasviel/Fooocus/issues/3090))

**Reply by MetatronR** [reply]
I would add a specific component for the performane-optimised options.
It would be selectable as a radio option, but also having a dropdown for selecting the desired method.

The best algorithm/model would be selected by default and also having the '(recommended)' suffix.
Other options could have suffixes like: fast, fastest, etc.

**Comment by wisdombox44** [comment]
I tried Hyper SD 12 steps Lora i got good results. Hyper SD 12 can be an option too.

**Comment by HelloLaohe** [comment]
I initially hoped to replace it with PCM, but even after 16 steps, the PCM still didn't achieve the desired results. After further testing, I felt that the Hyper-SD 12-step method, tested various sampling techniques, and provided comparisons. I hope to see better results by switching LCM to the Hyper-SD 12-step method and changing Lightning to 8 steps.

![1](https://github.com/lllyasviel/Fooocus/assets/131928833/d624cddd-6de9-4b95-b671-45c2b2004a9f)
![2](https://github.com/lllyasviel/Fooocus/assets/131928833/d5a2ed1e-32d4-4bcb-b0de-af170d033fa2)



           print("Enter Hyper-SD(12) mode.")
            progressbar(async_task, 1, "Downloading Hyper-SD(12) components ...")
            performance_loras += [(modules.config.downloading_sdxl_lcm_lora(), 1.0)]
            if refiner_model_name != "None":
              print(f"Refiner disabled in Hyper-SD(12)  mode.")
            refiner_model_name = "None"
            sampler_name = "euler_ancestral"
            scheduler_name = "ddim_uniform"
            sharpness = 1.0
            guidance_scale = 3.0
            adaptive_cfg = 8.0
            refiner_switch = 2.0
            adm_scaler_positive = 1.0
            adm_scaler_negative = 1.0
            adm_scaler_end = 0.0

**Comment by HelloLaohe** [comment]
The following are the images generated by using Hyper-SD 12-step method for reference.：
![2024-06-05_13-03-19_1892](https://github.com/lllyasviel/Fooocus/assets/131928833/314c19a9-d8dc-4762-88c1-9347bbcd0861)
![2024-06-05_13-03-45_9492](https://github.com/lllyasviel/Fooocus/assets/131928833/be008f17-932f-4955-b681-415291ab7f84)
![2024-06-05_13-04-20_8425](https://github.com/lllyasviel/Fooocus/assets/131928833/01a71f96-ecd0-4bb0-947e-857c478e688e)
![2024-06-05_13-04-39_5868](https://github.com/lllyasviel/Fooocus/assets/131928833/b6e26122-cd4a-4c8f-b33f-5d39d8c235ec)
![2024-06-05_13-04-56_9743](https://github.com/lllyasviel/Fooocus/assets/131928833/713aa0f3-4613-4f53-b11b-42f026863e19)
![2024-06-05_13-06-32_2359](https://github.com/lllyasviel/Fooocus/assets/131928833/1c95295b-9ee3-4ca4-8dad-41bd71764798)

**Comment by wisdombox44** [comment]
That is great for reference.
Lightning 8 step results look nice too. Sharp, vivid, clear

**Comment by HelloLaohe** [comment]
I hope to be able to add hpysd-sd 12 steps soon. Lightning should switch to 8 steps, otherwise, it's very practical to manually modify it every day's update.

![20240606122657](https://github.com/lllyasviel/Fooocus/assets/131928833/cd100239-a32a-40f2-884d-bef0df11e0b1)

**Reply by maolei133** [reply]
@HelloLaohe 你好，你的页面中那些其他功能是安装了那些插件？能说说吗？特别是中文显示插件。谢谢了

**Comment by jaimeastorga2000** [comment]
Remove LCM, change the Lightning LoRA from 4-step to 8-step.

Fooocus is supposed to be easy to use; too many options in the UI is confusing. Maybe hide it behind an advanced/debug tab.

Speaking of, do we really need both Lightning and Hyper-SD?

**Reply by HelloLaohe** [reply]
1, not everyone's graphics card is powerful, so it is necessary to have a way to generate images quickly. I am using a 3080ti, and some people around me are using a 2060 graphics card.
2, the modified hyper-sd (12 steps) is my default setting, and it also has very good effects. I hope to share it with more friends. Of course, I am also open to adding it to the advanced settings, and I can make the adjustments myself (modify the code a bit), but it's a pity that more people are not using it.

Here are my test images, using different content as much as possible.
![2024-06-12_08-51-26_8744](https://github.com/lllyasviel/Fooocus/assets/131928833/c88dc2b5-f457-43b3-a707-12fbb7510a11)
![2024-06-12_08-49-17_5251](https://github.com/lllyasviel/Fooocus/assets/131928833/b5463c13-5f3a-4953-8994-ff7124ec39b6)
![2024-06-12_08-49-52_2405](https://github.com/lllyasviel/Fooocus/assets/131928833/4975c891-bb74-40cc-85f3-233614f9acdc)
![2024-06-12_08-50-51_6626](https://github.com/lllyasviel/Fooocus/assets/131928833/7036daf6-a06f-4d0c-8ddf-d12b6f1f70d5)

**Comment by HelloLaohe** [comment]
----------------Modify the `async_worker.py` file----------------

            refiner_model_name = "None"
            sampler_name = "euler_ancestral"
            scheduler_name = "sgm_uniform"
            sharpness = 1.2
            guidance_scale = 3.0
            adaptive_cfg = 7.5
            refiner_switch = 2.0
            adm_scaler_positive = 1.0
            adm_scaler_negative = 1.0
            adm_scaler_end = 0.0


----------------Modify the `flags.py` file----------------

```
class PerformanceLoRA(Enum):
    QUALITY = None
    SPEED = None
    EXTREME_SPEED = "sdxl_lcm_lora.safetensors"   # (The model name remains the same, but the actual file is the hyper-sd 12-step version.) 
    LIGHTNING = "sdxl_lightning_4step_lora.safetensors"
    HYPER_SD = "sdxl_hyper_sd_4step_lora.safetensors"

class Steps(IntEnum):
    QUALITY = 60
    SPEED = 30
    EXTREME_SPEED = 12
    LIGHTNING = 8
    HYPER_SD = 4
```

**Reply by Zsural** [reply]
Hello!Can you tell me where to download hyper-sd 12-step version?I didn't find downloadlinks in modules.config.py,Thanks alot!

**Reply by mashb1t** [reply]
@Zsural here you go: https://huggingface.co/ByteDance/Hyper-SD/blob/main/Hyper-SDXL-12steps-CFG-lora.safetensors
Keep in mind that this is the CFG version and you have to adjust the guidance scale and weight accordingly

**Reply by Zsural** [reply]
@mashb1t Thank you!

**Comment by steppres** [comment]
This could be a chance to make some more substantial changes to the performance settings as whole. At the moment for a novice user the names for each setting are somewhat ~~misleading~~ confusing. Maybe we could rename the settings to something like this:

"Quality", "Normal", "Fast", "Very Fast", "Ultra Fast"

These would correspond to 60-step, 30-step, 12-step (Hyper-SD), 8-step (Lightning) and 4-step (Hyper-SD). 

It could also allow for the legacy LCM to to still be regenerated from old image logs, just not displayed in the UI. Just an idea.
